src/common/io.ts

```diff
@@ -220,7 +225,7 @@ export class ARIOReadable implements AoARIORead {
       },
     ];
 
-    return this.process.read<AoEpochData | undefined>({
+    return this.process.read<AoEpochData<AoEpochDistributionTotalsData>>({
```

Comment:
this should be `AoEpochData` with `distributions: AoEpochDistributionTotalsData`

Reply:
it is that, AoEpochData is a generic taking AoEpochDistributionTotalsData to populate on distributions. though the type declaration here is irrelevant to the caller as it'll get overwritten by the function declaration. its just giving the compiler some info to type check us on

Reply:
ah, i missed the use of the generic